samza-core/src/main/scala/org/apache/samza/system/SystemConsumers.scala

```diff
@@ -154,6 +164,11 @@ class SystemConsumers (
   metrics.setUnprocessedMessages(() => totalUnprocessedMessages)
 
   def start {
+    for ((systemStreamPartition, offset) <- sspToRegisteredOffsets.asScala) {
```

Comment:
Just wondering why is this necessary? When would consumers restart multiple times and seek to the last registered offset? Is it some error scenario you're handling?